### ISSUE TITLE
Design Intent - Accessibility Feedback - Education Data Migration, WEAMS Public

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#98857]
## Next Steps for the VFS team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://ows.io/qs/gb1dfnkl) at any time._

- [x] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [x] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

-

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [x] **Must:** Avoid using tabs, unless there's no other option. Tabs are visually intuitive on large screens, but they pose a number of accessibility challenges. Some examples:
   - For screen reader users, the interaction pattern for navigating a set of tabs is different than the pattern for navigating a set of links or other more common elements. We can't assume that screen reader users are *expert* screen reader users who are familiar with every interaction pattern, and some users may get stuck.
   - For low vision users using either browser zoom or a screen magnifier, tracking which tab is active can be difficult (either because tabs don't play well in high-zoom viewports, or because magnified content may not contain the tab controls).
   - For sighted voice command / speech recognition software users, some interactions require being able to identify the type of element on the page (eg. link, button, form input, etc). Tabs aren't typically styled to be visibly identifiable as a link or a button (and in some cases are coded as custom elements), and so these interaction patterns break down.
- [x] **Must:** Use headings to create structure on the page. The heading hierarchy makes content scannable for sighted users, and provides navigational landmarks for screen reader users skimming the page to find the section they're interested in. Looking at a [result details page](https://www.figma.com/design/xAydfsAlVVmhnFQA9MkeYw/Design-for-WEAMS-Public-in-CT-%26-SOB?node-id=2468-332829&t=YV6cSTDVfj4VTXt8-4), the "License" heading doesn't describe a section of content, instead providing a taxonomical description of what's on this page. And the sections of content that do exist (the Admin Info and Test Info that are grouped into tabs in the other version of the details page) don't have headings to differentiate them from each other.
- [x] **Must:** When you use a table for any purpose, make sure the table has a title (the `table-title` prop in the `va-table` component). As a screen reader user, navigating a table can be a huge pain, and the title provides a summary of what's included in the table before you have to start exploring it so you can decide if you want to skip it or dig into it.
- [x] **Must:** Be careful to use the right element for the right purpose, especially for links and buttons. At VA we practice material honesty --- buttons are coded as buttons, perform an action, and are visibly identifiable as buttons; links are coded as links, navigate the browser, and are visibly identifiable as links. I'm noticing the "Back to search" button on the [results page](https://www.figma.com/design/xAydfsAlVVmhnFQA9MkeYw/Design-for-WEAMS-Public-in-CT-%26-SOB?node-id=2233-243466&t=YV6cSTDVfj4VTXt8-4), for example.
- [x] **Must:** All repeated elements on a page (eg. the "View test amount details" links on the [results page](https://www.figma.com/design/xAydfsAlVVmhnFQA9MkeYw/Design-for-WEAMS-Public-in-CT-%26-SOB?node-id=2233-243466&t=YV6cSTDVfj4VTXt8-4)) must have unique accessible names, so that screen reader users can identify them without needing to explore the surrounding context. The easiest solution here is to use descriptive link text, eg. "View test amount details for [name of license]." When that's not desirable for design reasons, the next best option is to use the `label` prop on the `va-link` component to add an `aria-label`, which will be announced to screen reader users instead of the link text. That screen reader text must be descriptive and unique.
- [x] **Should:** Think about the order in which information is presented, especially for users who aren't able to perceive visual cues about the structure of the page or changes to content. If I'm a screen reader user, I may experience the content of the page in a very linear way --- one thing is announced, then another, then another, then another. If content is dynamically changed upstream from where I am now (eg. [adding the info alert when the state field changes](https://www.figma.com/design/xAydfsAlVVmhnFQA9MkeYw/Design-for-WEAMS-Public-in-CT-%26-SOB?node-id=2233-243585&t=YV6cSTDVfj4VTXt8-4)) or if content is presented after the natural end to the user flow (eg. [FAQs that come after the search button](https://www.figma.com/design/xAydfsAlVVmhnFQA9MkeYw/Design-for-WEAMS-Public-in-CT-%26-SOB?node-id=2233-243624&t=YV6cSTDVfj4VTXt8-4)), I may never realize that content exists. That may be acceptable for content that's not critical to completing a task, but it's worth spending some time thinking about the impact of what might be lost for some users.
- [x] **Should:** Try to make interactions as predictable as possible. I'm thinking in particular about the flow in which the state field changes based on a user input elsewhere in the form, and the modal interrupts the user to prompt them to change their responses. Neither of these are really predictable behaviors to a first-time user, and the cognitive load for parsing and understanding these interactions may be high for all users.
   - [x] **Consider:** There are a number of other experiences on VA.gov where a user completes a search and then has an opportunity to refine their search through filters, or in which the search results page tells them why no results could be returned. I don't know the exact design solution for what you're doing here, but I'd encourage you to explore some other alternatives.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone

**Comment by cassidy-beach** [comment]
**Must: Avoid using tabs, unless there's no other option.** Tabs are visually intuitive on large screens, but they pose a number of accessibility challenges.
No longer using tabs for details pages
![Image](https://github.com/user-attachments/assets/354e524f-acc5-4527-90d4-805463d0fda3)

**Comment by cassidy-beach** [comment]
**Must: Use headings to create structure on the page.** 
Now using headings to create structure on the page
![Image](https://github.com/user-attachments/assets/6ef959aa-6f51-48d0-bd1d-9ab58379e5bd)

**Comment by cassidy-beach** [comment]
**Must: When you use a table for any purpose, make sure the table has a title (the table-title prop in the va-table component).** 
Tables now have a title 
![Image](https://github.com/user-attachments/assets/083ddf55-cf6d-414c-b503-6f4dddbed78b)

**Comment by cassidy-beach** [comment]
**Must: Be careful to use the right element for the right purpose, especially for links and buttons.**
Went through the tool to ensure we're practicing material honesty

**Comment by cassidy-beach** [comment]
**Must: All repeated elements on a page (eg. the "View test amount details" links on the [results page](https://www.figma.com/design/xAydfsAlVVmhnFQA9MkeYw/Design-for-WEAMS-Public-in-CT-%26-SOB?node-id=2233-243466&t=YV6cSTDVfj4VTXt8-4)) must have unique accessible names**, so that screen reader users can identify them without needing to explore the surrounding context.
![Image](https://github.com/user-attachments/assets/7ee2a6f8-7827-41b3-a236-2168149dea6e)
![Image](https://github.com/user-attachments/assets/5704201d-852e-4b66-a90e-62bae79cf43f)

**Comment by cassidy-beach** [comment]
**Should: Think about the order in which information is presented**, especially for users who aren't able to perceive visual cues about the structure of the page or changes to content. 
User can change content after initial search and new information will appear at the top. Search is more linear now
![Image](https://github.com/user-attachments/assets/5e95960c-9226-47fd-9481-0c29c66b76ed)
